irobotclient/configuration_handler.py

```
@@ -0,0 +1,159 @@
+
+
+def _check_input_file_argument(args):
+    """
+    Strip leading slash from input argument to prevent double slashes in API request.
+
+    :param args: the command line arguments
+    """
+    if args.input_file.endswith('/') or os.path.isdir(args.input_file):
+        raise IrobotClientException(errno=errno.ECONNABORTED,
+                                    message="Cannot download entire directories at present.")
+
+    if args.input_file.startswith('/'):
+        args.input_file = args.input_file.lstrip('/')
+
+
+def _check_output_directory_argument(args):
+    """
+    Check if the output directory already exists and if it already contains files of the same name as the input file.
+
+    :return: n/a
+
+    """
+
+    try:
+        # Expand the output_dir argument so the full directory path can be used in the rest of the program.
+        args.output_dir = os.path.expanduser(args.output_dir)
+
+        # Add a trailing slash to indicate directory.
+        if not args.output_dir.endswith('/'):
+            args.output_dir = args.output_dir + '/'
+
+        # Split up the path string to obtain just the file name without extensions and full path.
+        file_name = os.path.splitext((os.path.split(args.input_file))[1])[0]
+
+        for dir_file in os.listdir(args.output_dir):
+
+            # Basic match; will return true if the directory file begins with the input_file string.
+            # TODO - Improve regex on checking whether input file already exists.
+            if re.match(os.path.splitext(dir_file)[0], file_name) and not args.force:
+                raise IrobotClientException(errno=errno.EEXIST, message="File already exists. Please use the "
+                                                                        "--force option to overwrite.")
+
+    except IrobotClientException:
+        raise
+    except OSError:
+        raise
+    except:
+        raise Exception("UNKNOWN ERROR: Failed to validate the output directory.")
```

This exception masks the actual exception that was raised  (e.g. if a `FileNotFoundException` was raised, this exception handler just turns that into an "UKNOWN ERROR", which does not help the user).